File: tests/machine/x/fs/group_by_having.fs
open System
open System.Text.Json

type Anon1 = {
    name: string
    city: string
}
type Anon2 = {
    city: obj
    num: obj
}
let people: obj list = [{ name = "Alice"; city = "Paris" }; { name = "Bob"; city = "Hanoi" }; { name = "Charlie"; city = "Paris" }; { name = "Diana"; city = "Hanoi" }; { name = "Eve"; city = "Paris" }; { name = "Frank"; city = "Hanoi" }; { name = "George"; city = "Paris" }]
let big: obj list = [ for gKey, gItems in [ for p in people do yield p ] |> List.groupBy (fun p -> p.city) |> List.filter (fun (gKey, gItems) -> let g = {| key = gKey; items = gItems |} in List.length g.items >= 4) do
    let g = {| key = gKey; items = gItems |}
    yield { city = g.key; num = List.length g.items } ]
printfn "%A" (JsonSerializer.Serialize(big))

File: tests/transpiler/x/clj/group_by_having.clj
(ns main)

(require 'clojure.set)

(defrecord People [name city])

(def people [{:name "Alice" :city "Paris"} {:name "Bob" :city "Hanoi"} {:name "Charlie" :city "Paris"} {:name "Diana" :city "Hanoi"} {:name "Eve" :city "Paris"} {:name "Frank" :city "Hanoi"} {:name "George" :city "Paris"}])

(def big (for [g (for [[k rows] (group-by :key (for [p people :let [k (:city p)]] {:key k :item p})) :let [g {:key k :items (map :item rows)}] :when (>= (count (:items g)) 4)] g)] {:city (:key g) :num (count (:items g))}))

(defn -main []
  (println ((fn json_str [x] (cond (map? x) (str "{" (clojure.string/join "," (map (fn [[k v]] (str "\"" (name k) "\":" (json_str v))) x)) "}") (sequential? x) (str "[" (clojure.string/join "," (map json_str x)) "]") (string? x) (pr-str x) :else (str x))) big)))

(-main)
